components/Footer.tsx

```diff
@@ -18,57 +22,55 @@ const getFeedbackUrl = (currentURL) => {
   return `${feedbackPageUrlBase}${feedbackPageQueryPart}`;
 };
 
-type Props = {
-  siteTitle: string;
-};
+function getNavChildren(navItem: NavItem) {
+  if (!navItem || !('children' in navItem)) {
+    return [];
+  }
+
+  return (navItem.children || [])
+    .filter((child): child is NonNullable<typeof child> => !!child)
+    .map((child) => ({
+      id: child.id,
+      name: child.page.title,
+      slug: child.page.urlPath,
+    }));
+}
 
-function Footer({ siteTitle }: Props) {
+function Footer() {
   const plan = usePlan();
   const pathname = usePathname();
   const generalContent = plan.generalContent || {};
   const theme = useTheme();
 
+  const { navigationTitle: siteTitle } = getMetaTitles(plan);
   const { fundingInstruments, otherLogos, footerStatement } = theme.settings;
   const t = useTranslations();
-  let navLinks = [];
-  let staticPages = [];
-
-  plan.footer.items?.forEach((navItem) => {
-    const children = navItem.children.length > 0 ? [] : null;
-    if (children) {
-      navItem.children.forEach((child) => {
-        children.push({
-          id: child.id,
-          name: child.page.title,
-          slug: child.page.urlPath,
-        });
-      });
-    }
-    navLinks.push({
-      id: navItem.id,
-      name: navItem.page.title,
-      slug: navItem.children.length > 0 ? undefined : navItem.page.urlPath,
-      children,
+
+  const navLinks = (plan.footer?.items || [])
+    .filter((navItem): navItem is NonNullable<typeof navItem> => !!navItem)
+    .map((navItem) => {
+      if (!navItem || !('id' in navItem)) {
+        return null;
+      }
+
+      return {
+        id: navItem.id,
+        name: navItem.page.title,
+        slug:
+          'children' in navItem && (navItem.children || []).length > 0
+            ? undefined
+            : navItem.page.urlPath,
+        children: getNavChildren(navItem),
+      };
     });
-  });
-
-  if (plan.staticPages) {
-    const topMenuPages = plan.staticPages.filter((page) => page.footer);
-    staticPages = topMenuPages.map((page, index) => ({
-      id: `s${index}`,
-      name: page.name,
-      slug: page.slug,
-    }));
-    navLinks = navLinks.concat(staticPages);
-  }
```

Comment:
@terotik @tituomin `staticPages` seemed to be legacy, I couldn't find it set anywhere or queried in GQL. Can you confirm this is alright to remove?

Reply:
I would say ok to remove